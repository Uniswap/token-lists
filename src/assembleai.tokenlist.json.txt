{
  "name": "Assemble AI Token List",
  "logoURI": "https://raw.githubusercontent.com/AssembleAI-crypto/assemble-tokenlist/main/logo.png",
  "keywords": ["ASM", "AssembleAI", "Assemble AI", "Assemble protocol", "AssembleProtocol"],
  "timestamp": "2025-06-26T15:00:00Z",
  "version": { "major": 1, "minor": 0, "patch": 0 },
  "tokens": [
    {
      "chainId": 1,
      "address": "0x2565ae0385659badcada1031db704442e1b69982",
      "name": "ASSEMBLE",
      "symbol": "ASM",
      "decimals": 18,
      "logoURI": "https://raw.githubusercontent.com/AssembleAI-crypto/assemble-tokenlist/main/logo.png"
    },
    {
      "chainId": 8453,
      "address": "0x3b53604113B5677291BFc0bc255379E7a796559b",
      "name": "ASSEMBLE",
      "symbol": "ASM",
      "decimals": 18,
      "logoURI": "https://raw.githubusercontent.com/AssembleAI-crypto/assemble-tokenlist/main/logo.png"
    }
  ]
}