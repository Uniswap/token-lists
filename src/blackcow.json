{
  "name": "Blackcow Token List",
  "logoURI": "https://raw.githubusercontent.com/tonykim01/blackcow/main/BKC%20Main.png",
  "keywords": [
    "blackcow",
    "meme",
    "crypto",
    "BKC"
  ],
  "timestamp": "2025-06-20T20:48:44.818433",
  "version": {
    "major": 1,
    "minor": 0,
    "patch": 0
  },
  "tokens": [
    {
      "chainId": 1,
      "address": "0x28a229b144Fa6Cd82D7d0425626Ff2D65dEF9640",
      "name": "Blackcow Token",
      "symbol": "BKC",
      "decimals": 18,
      "logoURI": "https://raw.githubusercontent.com/tonykim01/blackcow/main/BKC%20Main.png"
    }
  ]
}