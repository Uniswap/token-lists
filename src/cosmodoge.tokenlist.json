{
  "name": "CosmoDoge Token List",
  "logoURI": "https://raw.githubusercontent.com/CosmoDoge/cosmodoge-tokenlist/main/logo.png",
  "keywords": ["cosmodoge", "tokenlist", "bep20"],
  "timestamp": "2025-04-04T00:00:00+00:00",
  "tokens": [
    {
      "name": "CosmoDoge",
      "symbol": "CDOGE",
      "address": "0x6dEd7a840077bB7752A4d09b502e3e9bAE44dcD1",
      "chainId": 56,
      "decimals": 18,
      "logoURI": "https://raw.githubusercontent.com/CosmoDoge/cosmodoge-tokenlist/main/logo.png"
    }
  ],
  "version": {
    "major": 1,
    "minor": 0,
    "patch": 0
  }
}
