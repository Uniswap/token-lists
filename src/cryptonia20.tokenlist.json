{
"name": "Cryptonia Token List",
"timestamp": "2025-06-25T18:00:00Z",
"version": {
"major": 1,
"minor": 0,
"patch": 0
},
"tags": {},
"tokens": [
{
"chainId": 1,
"address": "0xFd655eB8fB0296DeeBFdFdc18a05a522186985e1",
"name": "Cryptonia20",
"symbol": "CT20",
"decimals": 18,
"logoURI": "https://tonsite.com/images/logo-ct20.png",
"extensions": {
"website": "https://popo20i.github.io/Cryptonia20/",
"social": {
"twitter": "https://twitter.com/20Cryptonia"
}
}
}
]
}
