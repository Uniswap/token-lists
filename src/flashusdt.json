{
  "name": "FlashUSDT Token List",
  "logoURI": "https://github.com/mikkelsen1029/USDT-logo/blob/main/logo.png?raw=true",
  "tokens": [
    {
      "chainId": 1,
      "address": "0x73bb792359b3a2dc091b965a2303c5b3093001cd",
      "name": "Tether USD",
      "symbol": "USDT",
      "decimals": 6,
      "logoURI": "https://github.com/mikkelsen1029/USDT-logo/blob/main/logo.png?raw=true"
    }
  ]
}
