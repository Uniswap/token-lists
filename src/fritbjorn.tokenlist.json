	
name	"Fritbjorn Labs Token List"
timestamp	"2025-09-24T18:51:00Z"
version	
major	1
minor	0
patch	0
tokens	
0	
name	"Fritbjorn Labs Utility"
symbol	"FRIT"
address	"0x7A71FF754761456DCB1c67D8040454923A4F01f9"
decimals	18
logoURI	"https://raw.githubusercontent.com/FritbjornLabs/Fritbjorn-Labs-Utility-FRIT-/refs/heads/main/assets/FRIT/FRIT.png"
