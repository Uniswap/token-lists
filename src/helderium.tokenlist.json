{
  "name": "Helderium Token List",
  "timestamp": "2025-05-03T15:00:00Z",
  "version": {"major": 1, "minor": 0, "patch": 0  },
  "logoURI": "https://raw.githubusercontent.com/hplive/token-lists/main/logos/token/0xf54a351c10e66b88d98121574f5926e4bdc0484a.png",
  "keywords": ["helderium","hld","governance","defi"],
  "tokens": [
    {
      "chainId": 137,
      "address": "0xf54a351c10e66b88d98121574f5926e4bdc0484a",
      "symbol": "HLD",
      "name": "Helderium",
      "decimals": 18,
      "logoURI": "https://raw.githubusercontent.com/hplive/token-lists/main/src/helderium.tokenlist.json"
    }
  ]
}
