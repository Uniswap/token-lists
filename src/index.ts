import binanceWormhole from './binance-wormhole.tokenlist.json';
import schema from './tokenlist.schema.json';


export const binanceWormholeList: TokenList = {
  name: 'Binance Wormhole Token List',
  timestamp: new Date().toISOString(),
  version: {
    major: 1,
    minor: 0,
    patch: 0,
  },
  tags: {},
  logoURI:'https://cyan-sophisticated-mockingbird-33.mypinata.cloud/ipfs/bafkreigbhd3guo7bznbhsdkkzzjrlyyv3swpr33vpsgkgoxzx5v26wvcpm?pinataGatewayToken=DDN9vdbXaKS-moumQZ8sSrnYGZ9TxfhUqCBxRd8waHjUTw5xqN0upk9b4m93g5ki',
  keywords: ['binance', 'wormhole'],
  tokens: [...binanceWormhole],
};


export default {
  binanceWormholeList: buildList(binanceWormholeList),
};



export * from './types';
export * from './isVersionUpdate';
export * from './getVersionUpgrade';
export * from './diffTokenLists';
export * from './minVersionBump';
export * from './nextVersion';
export * from './versionComparator';

export { schema };



