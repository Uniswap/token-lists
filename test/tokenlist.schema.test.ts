import Ajv from 'ajv';
import { schema } from '../src';
import exampleList from './schema/example.tokenlist.json';
import exampleNameSymbolSpecialCharacters from './schema/example-name-symbol-special-characters.tokenlist.json';
import bigExampleList from './schema/bigexample.tokenlist.json';
import exampleListMinimum from './schema/exampleminimum.tokenlist.json';
import emptyList from './schema/empty.tokenlist.json';
import emptyNameSymbol from './schema/empty-name-symbol.tokenlist.json';
import bigWords from './schema/bigwords.tokenlist.json';
import invalidTokenAddress from './schema/invalidtokenaddress.tokenlist.json';
import invalidTimestamp from './schema/invalidtimestamp.tokenlist.json';
import invalidLogoURI1 from './schema/invalidlogouri.1.tokenlist.json';
import invalidLogoURI2 from './schema/invalidlogouri.2.tokenlist.json';
import invalidVersion1 from './schema/invalidversion.1.tokenlist.json';
import invalidVersion2 from './schema/invalidversion.2.tokenlist.json';
import invalidVersion3 from './schema/invalidversion.3.tokenlist.json';
import invalidDecimals1 from './schema/invaliddecimals.1.tokenlist.json';
import invalidNumTags from './schema/invalidNumTags.tokenlist.json';
import invalidDecimals2 from './schema/invaliddecimals.2.tokenlist.json';
import extensionsValid from './schema/extensions-valid.tokenlist.json';
import extensionsInvalid from './schema/extensions-invalid.tokenlist.json';
import extensionsValidObject from './schema/extensions-valid-object.tokenlist.json';
import extensionsInvalidObjectTooDeep from './schema/extensions-invalid-object-too-deep.tokenlist.json';
import tokenSymbolWithPeriod from './schema/tokenwithperiodsymbol.tokenlist.json';
import crossChainExtensions from './schema/example-crosschain.tokenlist.json';
import addFormats from 'ajv-formats';

const ajv = new Ajv({ allErrors: true });
addFormats(ajv);
const validator = ajv.compile(schema);

describe('schema', () => {
  it('is valid', () => {
    expect(ajv.validateSchema(schema)).toEqual(true);
  });

  function checkSchema(schema: any, valid: boolean): void {
    const isValid = validator(schema);
    expect(validator.errors).toMatchSnapshot();
    expect(isValid).toEqual(valid);
  }

  it('works for example schema', () => {
    checkSchema(exampleList, true);
  });

  it('works for special characters schema', () => {
    checkSchema(exampleNameSymbolSpecialCharacters, true);
  });

  it('works for big example schema', () => {
    checkSchema(bigExampleList, true);
  });

  it('minimum example schema', () => {
    checkSchema(exampleListMinimum, true);
  });

  it('requires name, timestamp, version, tokens', () => {
    checkSchema({}, false);
  });

  it('empty list fails', () => {
    checkSchema(emptyList, false);
  });

  it('works for empty names and symbols', () => {
    checkSchema(emptyNameSymbol, true);
  });

  it('fails with big names', () => {
    checkSchema(bigWords, false);
  });

  it('checks token address', () => {
    checkSchema(invalidTokenAddress, false);
  });

  it('invalid timestamp', () => {
    checkSchema(invalidTimestamp, false);
  });

  it('invalid logo URI', () => {
    checkSchema(invalidLogoURI1, false);
    checkSchema(invalidLogoURI2, false);
  });

  it('invalid decimals', () => {
    checkSchema(invalidDecimals1, false);
    checkSchema(invalidDecimals2, false);
  });

  it('invalid number of tags on token', () => {
    checkSchema(invalidNumTags, false);
  });

  it('checks version', () => {
    checkSchema(invalidVersion1, false);
    checkSchema(invalidVersion2, false);
    checkSchema(invalidVersion3, false);
  });

  it('checks extensions', () => {
    checkSchema(extensionsValid, true);
    checkSchema(extensionsInvalid, false);
  });

  it('checks extensions with object', () => {
    checkSchema(extensionsValidObject, true);
  });

  it('checks extensions with too much nesting', () => {
    checkSchema(extensionsInvalidObjectTooDeep, false);
  });

  it('token symbols may contain periods', () => {
    checkSchema(tokenSymbolWithPeriod, true);
  });

  it('cross chain extensions example', () => {
    checkSchema(crossChainExtensions, true);
  });

  it('allows up to 10k tokens', () => {
    const exampleListWith10kTokens = {
      ...exampleList,
      tokens: [...Array(10000)].map(() => exampleList.tokens[0]),
    };
    checkSchema(exampleListWith10kTokens, true);
  });

  it('fails with 10001 tokens', () => {
    const exampleListWith10kTokensPlusOne = {
      ...exampleList,
      tokens: [...Array(10001)].map(() => exampleList.tokens[0]),
    };
    checkSchema(exampleListWith10kTokensPlusOne, false);
  });

  it('allows additional top-level fields', () => {
    const exampleListWithUnknownField = {
      ...exampleList,
      unknownField: 'foo',
    };
    checkSchema(exampleListWithUnknownField, true);
  });
});
