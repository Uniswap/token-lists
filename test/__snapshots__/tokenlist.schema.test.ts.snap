// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema allows additional top-level fields 1`] = `null`;

exports[`schema allows up to 10k tokens 1`] = `null`;

exports[`schema checks extensions 1`] = `null`;

exports[`schema checks extensions 2`] = `
Array [
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "type",
    "message": "must be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "type",
    "message": "must be object",
    "params": Object {
      "type": "object",
    },
    "schemaPath": "#/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/abc",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "type",
    "message": "must be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "type",
    "message": "must be object",
    "params": Object {
      "type": "object",
    },
    "schemaPath": "#/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/def",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "maxLength",
    "message": "must NOT have more than 42 characters",
    "params": Object {
      "limit": 42,
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/maxLength",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "type",
    "message": "must be object",
    "params": Object {
      "type": "object",
    },
    "schemaPath": "#/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/tooLong",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
]
`;

exports[`schema checks extensions with object 1`] = `null`;

exports[`schema checks extensions with too much nesting 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "type",
    "message": "must be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "type",
    "message": "must be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "type",
    "message": "must be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/0/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "type",
    "message": "must be boolean",
    "params": Object {
      "type": "boolean",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/1/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "type",
    "message": "must be number",
    "params": Object {
      "type": "number",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/2/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "type",
    "message": "must be null",
    "params": Object {
      "type": "null",
    },
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf/3/type",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/definitions/ExtensionPrimitiveValue/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1/anotherInvalidObject",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject/1",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
  Object {
    "instancePath": "/tokens/0/extensions/withObject",
    "keyword": "anyOf",
    "message": "must match a schema in anyOf",
    "params": Object {},
    "schemaPath": "#/anyOf",
  },
]
`;

exports[`schema checks token address 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/address",
    "keyword": "pattern",
    "message": "must match pattern \\"^0x[a-fA-F0-9]{40}$\\"",
    "params": Object {
      "pattern": "^0x[a-fA-F0-9]{40}$",
    },
    "schemaPath": "#/properties/address/pattern",
  },
]
`;

exports[`schema checks version 1`] = `
Array [
  Object {
    "instancePath": "/version/patch",
    "keyword": "minimum",
    "message": "must be >= 0",
    "params": Object {
      "comparison": ">=",
      "limit": 0,
    },
    "schemaPath": "#/definitions/Version/properties/patch/minimum",
  },
]
`;

exports[`schema checks version 2`] = `
Array [
  Object {
    "instancePath": "/version/minor",
    "keyword": "minimum",
    "message": "must be >= 0",
    "params": Object {
      "comparison": ">=",
      "limit": 0,
    },
    "schemaPath": "#/definitions/Version/properties/minor/minimum",
  },
]
`;

exports[`schema checks version 3`] = `
Array [
  Object {
    "instancePath": "/version/major",
    "keyword": "minimum",
    "message": "must be >= 0",
    "params": Object {
      "comparison": ">=",
      "limit": 0,
    },
    "schemaPath": "#/definitions/Version/properties/major/minimum",
  },
]
`;

exports[`schema cross chain extensions example 1`] = `null`;

exports[`schema empty list fails 1`] = `
Array [
  Object {
    "instancePath": "/tokens",
    "keyword": "minItems",
    "message": "must NOT have fewer than 1 items",
    "params": Object {
      "limit": 1,
    },
    "schemaPath": "#/properties/tokens/minItems",
  },
]
`;

exports[`schema fails with 10001 tokens 1`] = `
Array [
  Object {
    "instancePath": "/tokens",
    "keyword": "maxItems",
    "message": "must NOT have more than 10000 items",
    "params": Object {
      "limit": 10000,
    },
    "schemaPath": "#/properties/tokens/maxItems",
  },
]
`;

exports[`schema fails with big names 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/name",
    "keyword": "maxLength",
    "message": "must NOT have more than 60 characters",
    "params": Object {
      "limit": 60,
    },
    "schemaPath": "#/properties/name/maxLength",
  },
]
`;

exports[`schema invalid decimals 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/decimals",
    "keyword": "maximum",
    "message": "must be <= 255",
    "params": Object {
      "comparison": "<=",
      "limit": 255,
    },
    "schemaPath": "#/properties/decimals/maximum",
  },
]
`;

exports[`schema invalid decimals 2`] = `
Array [
  Object {
    "instancePath": "/tokens/0/decimals",
    "keyword": "minimum",
    "message": "must be >= 0",
    "params": Object {
      "comparison": ">=",
      "limit": 0,
    },
    "schemaPath": "#/properties/decimals/minimum",
  },
]
`;

exports[`schema invalid logo URI 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/logoURI",
    "keyword": "format",
    "message": "must match format \\"uri\\"",
    "params": Object {
      "format": "uri",
    },
    "schemaPath": "#/properties/logoURI/format",
  },
]
`;

exports[`schema invalid logo URI 2`] = `
Array [
  Object {
    "instancePath": "/logoURI",
    "keyword": "format",
    "message": "must match format \\"uri\\"",
    "params": Object {
      "format": "uri",
    },
    "schemaPath": "#/properties/logoURI/format",
  },
]
`;

exports[`schema invalid number of tags on token 1`] = `
Array [
  Object {
    "instancePath": "/tokens/0/tags",
    "keyword": "maxItems",
    "message": "must NOT have more than 10 items",
    "params": Object {
      "limit": 10,
    },
    "schemaPath": "#/properties/tags/maxItems",
  },
]
`;

exports[`schema invalid timestamp 1`] = `
Array [
  Object {
    "instancePath": "/timestamp",
    "keyword": "format",
    "message": "must match format \\"date-time\\"",
    "params": Object {
      "format": "date-time",
    },
    "schemaPath": "#/properties/timestamp/format",
  },
]
`;

exports[`schema minimum example schema 1`] = `null`;

exports[`schema requires name, timestamp, version, tokens 1`] = `
Array [
  Object {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'name'",
    "params": Object {
      "missingProperty": "name",
    },
    "schemaPath": "#/required",
  },
  Object {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'timestamp'",
    "params": Object {
      "missingProperty": "timestamp",
    },
    "schemaPath": "#/required",
  },
  Object {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'version'",
    "params": Object {
      "missingProperty": "version",
    },
    "schemaPath": "#/required",
  },
  Object {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'tokens'",
    "params": Object {
      "missingProperty": "tokens",
    },
    "schemaPath": "#/required",
  },
]
`;

exports[`schema token symbols may contain periods 1`] = `null`;

exports[`schema works for big example schema 1`] = `null`;

exports[`schema works for empty names and symbols 1`] = `null`;

exports[`schema works for example schema 1`] = `null`;

exports[`schema works for special characters schema 1`] = `null`;
